WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java

```diff
@@ -221,30 +260,26 @@ private void unpackIntent(@NonNull Intent intent) {
         }
     }
 
-    private boolean matchesInProgressMedia(final @NonNull MediaModel media) {
-        return mCurrentUpload != null && media.getLocalSiteId() == mCurrentUpload.getLocalSiteId();
-    }
-
-    private void cancelUpload() {
-        if (mCurrentUpload != null) {
-            dispatchCancelAction(mCurrentUpload);
+    private void cancelUpload(MediaModel oneUpload) {
+        if (oneUpload != null) {
+            dispatchCancelAction(oneUpload, mSiteStore.getSiteByLocalId(oneUpload.getLocalSiteId()));
```

Comment:
Let's also check if the returned site is `null` here, and print a log like we do [here](https://github.com/wordpress-mobile/WordPress-Android/pull/5383/files#diff-339e1ea3210c67f894a26e04d833f5f9R183) - and avoid `dispatchCancelAction()` in that case.